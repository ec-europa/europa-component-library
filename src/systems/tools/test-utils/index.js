const merge = require('deepmerge');
const twing = require('@ecl-twig/ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const test = document.createElement('test');
      test.innerHTML = html.trim();
      resolve(test);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
