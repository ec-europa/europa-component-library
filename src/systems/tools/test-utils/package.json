{
  "name": "@ecl-twig/test-utils",
  "version": "2.31.1",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
