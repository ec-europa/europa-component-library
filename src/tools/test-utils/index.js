const merge = require('deepmerge');
const twing = require('@ecl/twig-ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const div = document.createElement('div');
      div.innerHTML = html.trim();
      resolve(div.firstChild);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
