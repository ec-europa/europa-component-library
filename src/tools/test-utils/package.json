{
  "name": "@ecl/test-utils",
  "version": "3.4.2",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
