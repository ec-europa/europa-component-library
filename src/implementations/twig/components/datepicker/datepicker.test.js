import { merge, renderTwigFileAsNode } from '@ecl/test-utils';

import demoData from '@ecl/specs-component-datepicker/demo/data';

describe('Datepicker', () => {
  const template = '@ecl/datepicker/datepicker.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);

  test('renders correctly', () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });

  test('renders correctly with extra class names', () => {
    expect.assertions(1);

    const optionsWithExtraClasses = merge(demoData, {
      extra_classes: 'custom-class custom-class--test',
    });

    return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
  });

  test('renders correctly with extra attributes', () => {
    expect.assertions(1);

    const optionsWithExtraClasses = merge(demoData, {
      extra_attributes: [
        { name: 'data-test', value: 'data-test-value' },
        { name: 'data-test-1', value: 'data-test-value-1' },
      ],
    });

    return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
  });
});
