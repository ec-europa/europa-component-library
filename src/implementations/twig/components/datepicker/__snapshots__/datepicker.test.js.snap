// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Datepicker renders correctly 1`] = `
<div
  class="ecl-form-group"
>
  <label
    class="ecl-form-label"
    for="example-input-id-1"
  >
    Label    
    <span
      class="ecl-form-label__required"
    >
      *
    </span>
  </label>
  <div
    class="ecl-help-block"
  >
    This is the input's helper text.
  </div>
  <div
    class="ecl-datepicker"
  >
    <input
      class="ecl-datepicker__field ecl-text-input ecl-text-input--s"
      data-ecl-auto-init="Datepicker"
      data-ecl-datepicker-toggle=""
      id="example-input-id-1"
      name="example-input-id-1"
      placeholder="DD-MM-YYYY"
      required=""
      type="text"
      value="01-10-2020"
    />
    <svg
      aria-hidden="true"
      class="ecl-icon ecl-icon--s ecl-datepicker__icon"
      focusable="false"
    >
      <use
        xlink:href="/icons.svg#calendar"
      />
    </svg>
  </div>
</div>
`;

exports[`Datepicker renders correctly with extra attributes 1`] = `
<div
  class="ecl-form-group"
>
  <label
    class="ecl-form-label"
    for="example-input-id-1"
  >
    Label    
    <span
      class="ecl-form-label__required"
    >
      *
    </span>
  </label>
  <div
    class="ecl-help-block"
  >
    This is the input's helper text.
  </div>
  <div
    class="ecl-datepicker"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
    null=""
  >
    <input
      class="ecl-datepicker__field ecl-text-input ecl-text-input--s"
      data-ecl-auto-init="Datepicker"
      data-ecl-datepicker-toggle=""
      id="example-input-id-1"
      name="example-input-id-1"
      placeholder="DD-MM-YYYY"
      required=""
      type="text"
      value="01-10-2020"
    />
    <svg
      aria-hidden="true"
      class="ecl-icon ecl-icon--s ecl-datepicker__icon"
      focusable="false"
    >
      <use
        xlink:href="/icons.svg#calendar"
      />
    </svg>
  </div>
</div>
`;

exports[`Datepicker renders correctly with extra class names 1`] = `
<div
  class="ecl-form-group"
>
  <label
    class="ecl-form-label"
    for="example-input-id-1"
  >
    Label    
    <span
      class="ecl-form-label__required"
    >
      *
    </span>
  </label>
  <div
    class="ecl-help-block"
  >
    This is the input's helper text.
  </div>
  <div
    class="ecl-datepicker custom-class custom-class--test"
  >
    <input
      class="ecl-datepicker__field ecl-text-input ecl-text-input--s"
      data-ecl-auto-init="Datepicker"
      data-ecl-datepicker-toggle=""
      id="example-input-id-1"
      name="example-input-id-1"
      placeholder="DD-MM-YYYY"
      required=""
      type="text"
      value="01-10-2020"
    />
    <svg
      aria-hidden="true"
      class="ecl-icon ecl-icon--s ecl-datepicker__icon"
      focusable="false"
    >
      <use
        xlink:href="/icons.svg#calendar"
      />
    </svg>
  </div>
</div>
`;
